### DISCUSSION TITLE
.\main.go:13:56: undefined: core.RecordEvent

### DISCUSSION THREAD

**Discussion by TheSynt4x** [pocketbase/pocketbase#1445] [Q&A] [answered]
Hello! I've been trying to extend pocketbase with my own functionality but I ran into some issues. 

This is my code, that I have copied straight from the documentation.

```go
package main

import (
    "log"

    "github.com/pocketbase/pocketbase"
    "github.com/pocketbase/pocketbase/core"
)

func main() {
    app := pocketbase.New()

    app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordEvent) error {
        if e.Record.Collection().Name == "posts" {
            log.Println(e.Record) // still unsaved

            // overwrite the newly submitted "posts" record approved to false
            e.Record.Set("approved", false)
        }
    
        return nil
    })

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

Here's what I get whenever I try to run/build it: `.\main.go:13:56: undefined: core.RecordEvent`

If I remove the OnRecordBeforeCreateRequest event hook, it works just fine. And I'm importing core too. I am not sure what I'm doing wrong here. I'm not running it with CGO and this is my Go version: `go version go1.19.4 windows/amd64`

**Comment by ganigeorgiev** [comment, marked as answer]
There is a typo in the docs (I'll fix it later today).
The name of the event argument for `OnRecordBeforeCreateRequest` is `RecordCreateEvent`.

**Reply by TheSynt4x** [reply]
Thank you! It now works.
